src/main/sensors/compass.c

```diff
@@ -504,15 +511,17 @@ uint32_t compassUpdate(timeUs_t currentTimeUs)
         for (int axis = 0; axis < XYZ_AXIS_COUNT; axis++) {
             // DEBUG 0-2: magADC[X], magADC[Y], magADC[Z]
             DEBUG_SET(DEBUG_MAG_CALIB, axis, lrintf(mag.magADC[axis]));
-            // DEBUG 4-6: estimated magnetometer bias
+            // DEBUG 4-6: estimated magnetometer bias, increases above zero when calibration starts
             DEBUG_SET(DEBUG_MAG_CALIB, axis + 4, lrintf(compassBiasEstimator.b[axis]));
         }
         // DEBUG 3: norm / length of magADC, ideally the norm stays constant independent of the orientation of the quad
         DEBUG_SET(DEBUG_MAG_CALIB, 3, lrintf(sqrtf(sq(mag.magADC[X]) + sq(mag.magADC[Y]) + sq(mag.magADC[Z]))));
         // map adaptive forgetting factor lambda from (lambda_min, 1.0f) -> (0, 2000)
-        const float mapLambdaGain = 1.0f / (1.0f - compassBiasEstimator.lambda_min + 1.0e-6f) * 2.0e3f;
         // DEBUG 7: adaptive forgetting factor lambda, after the transient phase it should converge to 2000
-        DEBUG_SET(DEBUG_MAG_CALIB, 7, lrintf((compassBiasEstimator.lambda - compassBiasEstimator.lambda_min) * mapLambdaGain));
+        float mapLambdaGain = 1.0f / (1.0f - compassBiasEstimator.lambda_min + 1.0e-6f) * 2.0e3f;
+        // force to zero to signal point of  completion of calibration run in sensors tab
+        mapLambdaGain = (magCalTimeRemaining > 0) ? (compassBiasEstimator.lambda - compassBiasEstimator.lambda_min) * mapLambdaGain : 0.0f;
```

Comment:
Only first (originaly `const`) value is `mapLambdaGain`. Reusing variable for other value is confusing. You can easily declare fresh variable, gcc is smart enough to reuse memory/registers.

Reply:
True.  Bad habit of mine.  I'll use two const variables.